torch/fx/operator_schemas.py

```diff
@@ -126,9 +126,9 @@ def is_homogeneous_int_tuple(t):
             return True
         return all(c is int or (c is Ellipsis) for c in contained)
 
-    if signature_type is List[int] and is_homogeneous_int_tuple(argument_type):
+    if signature_type is List[int]:
         # Tuple[int] is accepted for List[int] parameters
-        return True
+        return is_homogeneous_int_tuple(argument_type)
```

Comment:
Note that this function still doesn't accept lists, but I think to do that it would have to be changed completely. There simply isn't enough information without iterating through the actual list to check if they are all `int`.